src/flask_endpoint.py

```diff
@@ -100,22 +93,23 @@ def hpf_scoring():
                     "package_to_topic_dict": package_to_topic_dict,
                 }
             response_json.append(output_json)
+        return flask.jsonify(response_json)
     else:  # pragma: no cover
         current_app.logger.error(
             'No scoring region provided. HPF_SCORING_REGION is {}'.format(HPF_SCORING_REGION))
         response_json.append(
             {"Error": "No scoring region provided"})
-    return flask.jsonify(response_json)
+        return flask.jsonify(response_json), 501
```

Comment:
The 501 code is IIRC only applicable to HTTP methods (ex. sending a POST to a GET only endpoint).